bentoml/repository/__init__.py

```diff
@@ -24,58 +24,83 @@
 from bentoml import config
 from bentoml import archive
 from bentoml.exceptions import BentoMLRepositoryException
+from bentoml.utils.s3 import is_s3_url
+from bentoml.utils import Path
 
 
-# Bento, is a file format representing a saved BentoService
 @add_metaclass(ABCMeta)
-class BaseRepository(object):
+class BentoRepositoryBase(object):
+    """
+    BentoRepository is the interface for managing saved Bentos over file system or
+    cloud storage systems.
+
+    A Bento is a BentoService serialized into a standard file format that can be
+    easily load back to a Python session, installed as PyPI package, or run in Conda
+    or docker environment with all dependencies configured automatically
+    """
+
     @abstractmethod
-    def add(self, bento_service_instance, version=None):
+    def add(self, bento_service):
         """
         Adding a BentoService instance to target repository - this will resolve in a
-        call to BentoService#save, which creates a new Bento that contains all
+        call to BentoService#save_to_dir, which creates a new Bento that contains all
         serialized artifacts and related source code and configuration. Repository
         implementation will take care of storing and retriving the Bento files.
+        Return value is an URL(file path or s3 path for example), pointing to the
+        saved Bento file directory
         """
 
     @abstractmethod
-    def delete(self, bento_name, bento_version):
-        """
-        Deleting the Bento files that was added earlier
-        """
-
-    @abstractmethod
-    def get(self, bento_name, bento_version=None):
+    def get(self, bento_name, bento_version):
         """
         Get a file path to the saved Bento files, path must be accessible form local
         machine either through NFS or pre-downloaded to local machine
         """
 
     @abstractmethod
-    def list(self, bento_name=None):
+    def dangerously_delete(self, bento_name, bento_version):
         """
-        List all saved versions of a given Bento name
+        Deleting the Bento files that was added to this repository earlier, this may
+        break existing deployments or create issues when doing deployment rollback
         """
 
 
-class LocalRepository(BaseRepository):
-    def __init__(self, base_path):
-        self.base_path = base_path
+class _LocalBentoRepository(BentoRepositoryBase):
+    def __init__(self, base_url):
+        if not os.path.exists(base_url):
+            raise BentoMLRepositoryException(
+                "Local path '{}' not found when " "initializing local Bento repository"
```

Comment:
I assume we want to show the user what local path?  I am sure you might just fogot put `.format` on